aries_cloudagent/anoncreds/revocation.py

```diff
@@ -86,6 +90,29 @@ def profile(self) -> AskarProfile:
 
     # Revocation artifact management
 
+    async def _update_rev_list_record(
+        self, txn: AskarProfileSession, state, rev_reg_def_id: str
+    ):
+        entry = await txn.handle.fetch(
+            CATEGORY_REV_LIST,
+            rev_reg_def_id,
+            for_update=True,
+        )
+        if not entry:
+            raise AnonCredsRevocationError(
+                f"{CATEGORY_REV_LIST} with rev_reg_def_id "
+                f"{rev_reg_def_id} could not be found"
+            )
+
+        tags = entry.tags
+        tags["state"] = state
+        await txn.handle.insert(
+            CATEGORY_REV_LIST,
+            rev_reg_def_id,
+            value=entry.value,
+            tags=tags,
+        )
+
```

Comment:
The rev list record does need a state; however, the state of all anoncreds objects including the rev list can only be one of `failed`, `finished`, or `wait`. Whether a registry is active or not is not reflected as a `state` but with the separate `active` boolean tag (the code you added for this in `set_active_registry` looks okay, minus the `state` being set to a value other than `failed`, `finished`, or `wait`)

The posted and pending values on the rev list should be full, serialized revocation list objects. When no credentials have been revoked since the last publish to the anoncreds ledger, pending will be `None`. When `mark_pending_revocation` is called and no `pending` is set yet, the `posted` list is copied, `update_list` is called on the copy, then the copy is stored as `pending`. If `mark_pending_revocation` is called and pending is set, `update_list` is called on `pending` and then saved back to the record as `pending`.

When storing a `CATEGORY_REV_LIST` record, it will look something like:

```python
            async with self.profile.session() as session:
                await session.handle.insert(
                    CATEGORY_REV_LIST,
                    rev_reg_def_id,
                    {
                        "posted": result.revocation_list_state.revocation_list.to_json(),
                        "pending": None,
                    }
                )
```
Or:
```python
            async with self.profile.session() as session:
                await session.handle.insert(
                    CATEGORY_REV_LIST,
                    rev_reg_def_id,
                    {
                        "posted": posted.to_json(),
                        "pending": pending.to_json(),
                    }
                )
```